src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1073,7 +1073,21 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
             if (implicitImpl.ContainsTupleNames() && MemberSignatureComparer.ConsideringTupleNamesCreatesDifference(implicitImpl, interfaceMember))
             {
                 // it is ok to implement implicitly with no tuple names, for compatibility with C# 6, but otherwise names should match
-                diagnostics.Add(ErrorCode.ERR_ImplBadTupleNames, implicitImpl.Locations[0], implicitImpl, interfaceMember);
+                diagnostics.Add(ErrorCode.ERR_ImplBadTupleNames, GetDiagnosticLocation(), implicitImpl, interfaceMember);
+
+                Location GetDiagnosticLocation()
+                {
+                    if (implicitImpl.ContainingType == implementingType)
+                    {
+                        return implicitImpl.Locations[0];
+                    }
+                    else
+                    {
+                        var @interface = interfaceMember.ContainingType;
+                        SourceMemberContainerTypeSymbol snt = implementingType as SourceMemberContainerTypeSymbol;
+                        return snt.GetImplementsLocation(@interface) ?? implementingType.Locations[0];
```

Comment:
>snt. [](start = 31, length = 4)

It feels like this access should be conditional because ```snt``` can be null. We should fix the other call site as well

Reply:
This could be rewritten as `return (implementingType as SourceMemberContainerTypeSymbol)?.GetImplementsLocation(@interface) ?? implementingType.Locations[0]`